[cmake] [unittests] removing scripts for Xcode project test targets

oozoofrog opened the pull request:
Some scripts in the add_swift_unittest function modifies LINK_LIBRARIES for include gtest, gtest_main libs. as a result lose link information that receving from add_unittest. 
This PR removes that script, so the test target no longer throws a "fatal error: 'gtest/gtest.h' file not found" from build test targets.

it had been affected by after https://reviews.llvm.org/D86616

oozoofrog committed:
[cmake] [unittests] include googletest headers to Xcode project header search paths

add target include directories on AddSwiftUnittests.cmake module.
currently generated Xcode project doesn't include googletest headers after https://reviews.llvm.org/D86616

rollmind commented:
@varungandhi-apple  check please, and thank you.

LucianoPAlmeida commented:
Hum, interesting ... this could potentially fix the problem people were having https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399 ?

compnerd commented on cmake/modules/AddSwiftUnittests.cmake:
```diff
@@ -10,6 +10,9 @@ function(add_swift_unittest test_dirname)
 
+  target_include_directories(${test_dirname} PUBLIC ${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
+  target_include_directories(${test_dirname} PUBLIC ${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
+
```
This is incorrect: https://github.com/llvm/llvm-project/blob/main/llvm/cmake/modules/AddLLVM.cmake#L1471

`add_unittest` already links against the library, which should have the public include directories specified, so that would already setup the include paths properly.

rollmind replied:
@compnerd thank you.
now I'm know this should be include header paths too, some reason Xcode project of llvm-project include header but swift Xcode project. i'm continuing to figure out the script.

ps. include_directories is working well too, but want to find more elegant way.

rollmind replied:
I suspect this is only issue for generating Xcode project so now testing some conditional options.

rollmind replied:
@compnerd now, i understand cmake build dependencies from this area. so removing some codes more.

compnerd commented on cmake/modules/AddSwiftUnittests.cmake:
```diff
@@ -10,6 +10,9 @@ function(add_swift_unittest test_dirname)
   # function defined by AddLLVM.cmake.
   add_unittest(SwiftUnitTests ${test_dirname} ${ARGN})
 
+  target_include_directories(${test_dirname} PUBLIC ${LLVM_MAIN_SRC_DIR}/utils/unittest/googletest/include)
+  target_include_directories(${test_dirname} PUBLIC ${LLVM_MAIN_SRC_DIR}/utils/unittest/googlemock/include)
```
`PUBLIC` is definitely not the right thing here.  The unit tests do not define the interfaces for gtest or gmock.

abdulajet commented:
> Hum, interesting ... this could potentially fix the problem people were having https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399 ?

FYI, I ran into this issue with a fresh install of macOS 10.15.7 and Xcode 12.3

rollmind commented:
> > Hum, interesting ... this could potentially fix the problem people were having https://forums.swift.org/t/gtest-gtest-h-not-found-in-typeref-cpp-while-compiling-the-compiler/44399 ?
> 
> FYI, I ran into this issue with a fresh install of macOS 10.15.7 and Xcode 12.3

sadly, build with Xcode 12.2 not working too.

oozoofrog committed:
remove unnecessary scripts

gtest, gtest_main LINK_LIBRARIES dependencies changed by that removed scripts to absolute library file path. as a result losing necessary include path dirs.

oozoofrog reviewed (commented):
@compnerd remove target_include_directories and unnecessary scripts. @varungandhi-apple @shahmishal @LucianoPAlmeida @abdulajet

compnerd commented:
@swift-ci please test

compnerd commented:
@swift-ci please test Linux platform

compnerd commented:
@swift-ci please test macOS platform

compnerd reviewed (approved):
LGTM

LucianoPAlmeida commented:
> @compnerd remove target_include_directories and unnecessary scripts. @varungandhi-apple @shahmishal @LucianoPAlmeida @abdulajet

@rollmind Perhaps the PR title and description should be updated to reflect the new changes :)

rollmind commented:
> > @compnerd remove target_include_directories and unnecessary scripts. @varungandhi-apple @shahmishal @LucianoPAlmeida @abdulajet
> 
> @rollmind Perhaps the PR title and description should be updated to reflect the new changes :)

I see, how about that changed title and description.

rollmind commented:
@compnerd Thanks

rollmind commented:
@varungandhi-apple @shahmishal 
Can i ask merge please?

varungandhi-apple commented:
@swift-ci test and merge

rollmind commented:
@varungandhi-apple Thnx! :)
